sretoolbox/container/image.py

```diff
@@ -218,6 +233,8 @@ def _parse_image_url(image_url):
             r'(?P<repository>[\w\-]+)?'  # Repository (optional)
             r'(?(repository)(?P<repo_slash>/))'  # Slash, if repo is present
             r'(?P<image>[\w\-.]+)'  # Image path (mandatory)
+            r'(?P<by_digest>@sha256:)?'  # Present iff it's a by-digest URI
+            r'(?(by_digest)(?P<digest>[0-9a-f]{64}))'  # Digest SHA: 64 lowercase hex chars
```

Comment:
Awesome, thanks for that!
I'd reccomend using the `@` as the marker and add the `sha256:` to the `<digest>` group. Then use the `<digest>` group to conditionally-negativelly look for the `tag_colon` ahead.

My suggestion here is:

```diff
-            r'(?P<by_digest>@sha256:)?'  # Present iff it's a by-digest URI
-            r'(?(by_digest)(?P<digest>[0-9a-f]{64}))'  # Digest SHA: 64 lowercase hex chars
-            r'(?P<tag_colon>:)?'  # Tag colon (optional)
-            r'(?(tag_colon)(?P<tag>[\w\-.]+))'  # Tag (if tag colon is present)
+            # Digest marker "@" (optional)
+            r'(?P<digest_at>@)?'
+            # Digest (if @ is present)
+            r'(?(digest_at)(?P<digest>sha256:[0-9a-f]{64}))'
+            # Tag colon (if digest not is present, optional)
+            r'(?(digest)|(?P<tag_colon>:))?'
+            # Tag (if tag colon is present)
+            r'(?(tag_colon)(?P<tag>[\w\-.]+))'
```

Reply:
Dig, will do. I’ll add a test to prove you can’t put a tag after a digest as well, which I think you could probably do as written.

Reply:
> I’ll add a test to prove you can’t put a tag after a digest as well, which I think you could probably do as written.

Well, this isn't strict enough for such a test to make sense, and this isn't the PR to start adding such strictness. Maybe later :)

Reply:
Sounds good